internal/telemetry/appinsights.go

```diff
@@ -0,0 +1,302 @@
+	CreateMssqlGetTagsHelp          = "Sqlcmd.Create.Mssql.GetTags.help"
+	CreateMssqlAcceptEula           = "Sqlcmd.Create.Mssql.AcceptEula"
+	CreateMssqlArchitecture         = "Sqlcmd.Create.Mssql.Architecture"
+	CreateMssqlCached               = "Sqlcmd.Create.Mssql.Cached"
+	CreateMssqlCollation            = "Sqlcmd.Create.Mssql.Collation"
+	CreateMssqlContextName          = "Sqlcmd.Create.Mssql.ContextName.c"
+	CreateMssqlErrorlogWaitLine     = "Sqlcmd.Create.Mssql.ErrorlogWaitLine"
+	CreateMssqlHostname             = "Sqlcmd.Create.Mssql.Hostname"
+	CreateMssqlName                 = "Sqlcmd.Create.Mssql.Name"
+	CreateMssqlOs                   = "Sqlcmd.Create.Mssql.Os"
+	CreateMssqlPasswordEncryption   = "Sqlcmd.Create.Mssql.PasswordEncryption"
+	CreateMssqlPasswordLength       = "Sqlcmd.Create.Mssql.PasswordLength"
+	CreateMssqlPasswordMinNumber    = "Sqlcmd.Create.Mssql.PasswordMinNumber"
+	CreateMssqlPasswordMinSpecial   = "Sqlcmd.Create.Mssql.PasswordMinSpecial"
+	CreateMssqlPasswordMinUpper     = "Sqlcmd.Create.Mssql.PasswordMinUpper"
+	CreateMssqlPasswordSpecialChars = "Sqlcmd.Create.Mssql.PasswordSpecialChars"
+	CreateMssqlPort                 = "Sqlcmd.Create.Mssql.Port"
+	CreateMssqlRegistry             = "Sqlcmd.Create.Mssql.Registry"
+	CreateMssqlRepo                 = "Sqlcmd.Create.Mssql.Repo"
+	CreateMssqlTag                  = "Sqlcmd.Create.Mssql.Tag"
+	CreateMssqlUserDatabase         = "Sqlcmd.Create.Mssql.UserDatabase.u"
+	CreateMssqlUsing                = "Sqlcmd.Create.Mssql.Using"
+
+	DeleteHelp  = "Sqlcmd.Delete.help"
+	DeleteForce = "Sqlcmd.Delete.Force"
+	DeleteYes   = "Sqlcmd.Delete.Yes"
+
+	Help = "Sqlcmd.help"
+
+	OpenHelp    = "Sqlcmd.Open.help"
+	OpenAdsHelp = "Sqlcmd.Open.Ads.help"
+
+	QueryHelp     = "Sqlcmd.Query.help"
+	QueryDatabase = "Sqlcmd.Query.database.d"
+	Queryquery    = "Sqlcmd.Query.query.q"
+	QueryText     = "Sqlcmd.Query.Text.t"
+	StartHelp     = "Sqlcmd.Start.help"
+	StopHelp      = "Sqlcmd.Stop.help"
+)
+
+func initCommonProps() {
+	if userId == "" {
+		userId = getOrCreateUserId()
+	}
+	if sessionId == "" {
+		sessionId = uuid.NewString()
+	}
+	locLang = localizer.LocaleName
+	userOs = runtime.GOOS
+}
+
+func init() {
+	telemetryEnabled := strings.ToLower(os.Getenv("SQLCMD_TELEMETRY"))
+	if telemetryEnabled == "false" {
+		isTelemetryEnabled = "false"
+	}
+	if isTelemetryEnabled == "true" {
+		InitializeAppInsights()
+		initCommonProps()
+	}
+}
+
+func InitializeTelemetryLogging() {
+	loggingEnabled := strings.ToLower(os.Getenv("SQLCMD_TELEMETRY_LOGGING"))
+	if isTelemetryEnabled == "true" && loggingEnabled == "true" {
+		// Add a diagnostics listener for printing telemetry messages
+		appinsights.NewDiagnosticsMessageListener(func(msg string) error {
+			fmt.Printf("[%s] %s\n", time.Now().Format(time.UnixDate), msg)
+			return nil
+		})
+	}
+}
+
+func InitializeAppInsights() {
+	instrumentationKey := "f305b208-557d-4fba-bf06-25345c4dfdbc"
```

Comment:
>"f305b208-557d-4fba-bf06-25345c4dfdbc" [](http://example.com/codeflow?start=23&length=38)

can we parameterize this key at build time? 
It may be desirable to have one key for local builds and a separate production key for our signed builds.